packages/zql/src/ivm/memory-source.ts

```diff
@@ -602,12 +611,12 @@ export function* generateWithOverlay(
   rows: Iterable<Row>,
   constraint: Constraint | undefined,
   overlay: Overlay | undefined,
-  connectionIndex: number,
+  lastPushedEpoch: number,
   compare: Comparator,
   filterPredicate?: (row: Row) => boolean | undefined,
 ) {
   let overlayToApply: Overlay | undefined = undefined;
-  if (overlay && connectionIndex <= overlay.outputIndex) {
```

Comment:
if the connection is less than or equal to the current output index, we should apply the overlay because that connection has been pushed to.